ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1424,3 +1437,35 @@ void GCS_MAVLINK_Copter::send_wind() const
         wind.length(),
         wind.z);
 }
+
+/*
+  send HIGH_LATENCY2 message
+ */
+void GCS_MAVLINK_Copter::send_high_latency()
+{
+    AP_AHRS &ahrs = AP::ahrs();
+    struct Location global_position_current;
+    UNUSED_RESULT(ahrs.get_position(global_position_current));
+
+    uint16_t target_altitude = 0;
+    uint16_t target_heading = 0;
+    uint16_t waypoint_distance = 0;
+    uint8_t target_airspeed = 0;
+
+    if (copter.ap.initialised) {
+        //const Vector3f &targets = copter.attitude_control->get_att_target_euler_cd();
+        const Mode *flightmode = copter.flightmode;
+        target_altitude = (global_position_current.alt * 0.01f) + (copter.pos_control->get_alt_error() * 1.0e-2f);
+        target_heading = flightmode->wp_bearing() * 1.0e-2f;
+        waypoint_distance = MIN(flightmode->wp_distance() * 1.0e-2f, UINT16_MAX);
+        target_airspeed = copter.pos_control->get_desired_velocity().length() * 1.0e-2f;
+    }
+
+    Vector3f airspeed_vec_bf;
+    Vector3f wind;
+    if (AP::ahrs().airspeed_vector_true(airspeed_vec_bf)) {
+        wind = AP::ahrs().wind_estimate();
+    }
+
+    send_high_latency_base((target_heading % 360), waypoint_distance, target_airspeed, target_altitude, wind.length(), degrees(atan2f(-wind.y, -wind.x)), 0);
```

Comment:
This is upside-down.  Have a look at how we send `GLOBAL_POSITION_INT` and `VFR_HUD` - the vehicles override getters for values the explicitly need to specify a value for.

Reply:
Fixed